src/xcode/ENA/ENA/Source/Models/DetectionMode.swift

```diff
@@ -28,8 +28,8 @@ enum DetectionMode {
 }
 
 extension DetectionMode {
-	static func from(backgroundStatus: UIBackgroundRefreshStatus) -> DetectionMode {
-		switch backgroundStatus {
+	static func fromBackgroundRefreshStatus() -> DetectionMode {
```

Comment:
IMHO this change would make the method harder to test, as you would need to swizzle/mock the `backgroundRefreshStatus` of `UIApplication.shared` when you'd like to test if it returns the expected values.

What do you think about moving it into the method signature as a default argument?

```swift
static func from(backgroundStatus: UIBackgroundRefreshStatus = UIApplication.shared.backgroundRefreshStatus) { ... }
```

This way you would get rid of the duplicated code you wanted to remove, but you could even add tests for this method.

Reply:
This was my first idea, but `DetectionMode.from()` looks a little bit strange.

Reply:
Fair point, didn't think about that. But what's about?

```swift
static func fromBackgroundStatus(_ backgroundStatus: UIBackgroundRefreshStatus = UIApplication.shared.backgroundRefreshStatus) { ... }
```

This should solve it, no? :)

Reply:
I kept the old method (for mocking) and the new one. Does this look better for you?